### PR TITLE
[PKG-4436] 0.28.1

### PR DESCRIPTION
**Pull request by ViridianMelody**
modin 0.28.1

**Destination channel:** defaults

### Links

- [PKG-4436] 
- [Upstream repository](https://github.com/modin-project/modin/tree/0.28.1)
- [Upstream changelog/diff](https://github.com/modin-project/modin/releases/tag/0.28.1)
- [0.26.1 PR](https://github.com/AnacondaRecipes/modin-feedstock/pull/11)

### Explanation of changes:

- This branch is based on 0.26.1.
- Small dependency pinning updates.

[PKG-4436]: https://anaconda.atlassian.net/browse/PKG-4436?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/update_versioneer.patch [hunk 1]
```diff
@@ -0,0 +1,2641 @@
...
+index 8a444639..b263a006 100644
+--- a/modin/__init__.py
++++ b/modin/__init__.py
+@@ -66,3 +66,6 @@ def set_execution(
+ 
+ __version__ = get_versions()["version"]
+ del get_versions
++
++from . import _version
++__version__ = _version.get_versions()['version']
```
Comment (Callek) [inline recipe/update_versioneer.patch:11]: Same comment from the 0.26.1 versioneer patch

**Review by Callek — APPROVED** [review]
Approved as far as versioneer is concerned.  See the comments on 0.26.1 as they apply here as well.

**Review by skupr-anaconda — APPROVED** [review]

File: recipe/meta.yaml [hunk 2]
```diff
@@ -68,25 +78,43 @@ outputs:
...
+        - xarray >=2022.03.0
+        - pyarrow >=7.0.0
+        - s3fs >=2022.05.0
+        - numexpr <2.8.5  # [py<312]
+
       commands:
-        - pip check
-        - python -c "from distributed import Client; import modin.pandas as pd; import numpy as np; frame_data = np.random.randint(0, 100, size=(2**10, 2**8)); df = pd.DataFrame(frame_data); print(df)"  # [not (linux and ppc64le)]
+        # Skip pip check on win due to strict dependency in docker-py
+        - pip check  # [not win]
```
Comment (boldorider4) [inline recipe/meta.yaml:107]: doesn't this mean that we should skip windows altogether, since the import test fails?

Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml:107]: This is not an import test. This test that pip check is happy, or in other words, it checks that each installed python package have their requirements satisfied. If this fail, it means we probably diverged from the upstream requirements.

Comment (ViridianMelody) [reply recipe/meta.yaml:107]: To add on a bit more detail, the issue was in the additional test dependencies. To ensure coverage, I added pip check to the `modin` output, which passes fine.

File: recipe/update_versioneer.patch (hunk 1 above)
Comment (lorepirri) [inline recipe/update_versioneer.patch:11]:
```suggestion
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -68,25 +78,43 @@ outputs:
...
         - modin.pandas
       requires:
-        - distributed
         - pip
+        - distributed >=2.22.0
+        - dask >=2.22.0
+        - pytest
+        - boto3
+        - requests
+        - matplotlib-base >=3.6.1
```
Comment (boldorider4) [inline recipe/meta.yaml:99]:
```suggestion
        - matplotlib-base >=3.6.3
```
https://github.com/modin-project/modin/blob/eac21a8463d7152b3ad48a8b80d6da5ac645be6a/environment-dev.yml#L30

File: recipe/meta.yaml [hunk 4]
```diff
@@ -68,25 +78,43 @@ outputs:
...
+        - xarray >=2022.03.0
```
Comment (boldorider4) [inline recipe/meta.yaml:100]:
```suggestion
        - xarray >=2022.12.0
```
https://github.com/modin-project/modin/blob/eac21a8463d7152b3ad48a8b80d6da5ac645be6a/environment-dev.yml#L23

File: recipe/meta.yaml [hunk 5]
```diff
@@ -68,25 +78,43 @@ outputs:
...
+        - pyarrow >=7.0.0
+        - s3fs >=2022.05.0
```
Comment (boldorider4) [inline recipe/meta.yaml:102]:
```suggestion
        - s3fs >=2022.11.0
```
https://github.com/modin-project/modin/blob/eac21a8463d7152b3ad48a8b80d6da5ac645be6a/requirements-dev.txt#L17

**Review by JeanChristopheMorinPerso — APPROVED** [review]
@boldorider4 comments could potentially be addressed in a future PR. I think this is good to go.

**Pull request merged**